### ISSUE TITLE
0.5.0 crashes on opening

### ISSUE THREAD

**Issue by C6H5-NO2** [HurryPeng/FragMem#2]
Tested on Mix2s with `MIUI 10 Global 8.6.26 Beta`.
```
java.lang.RuntimeException: Unable to start activity Componentlnfo{cc.hurrypeng.fragmem/ cc.hurrypeng.fragmem. MainActivity}:
java.lang. N ullPointerException: Attempt to invoke interface method ‘java.lang.Object[] java.util.Collection.toArray’ on a null object reference
at
android. app.ActivityThread. perform LaunchActivity(ActivityThr ead.java:2853)
at
android.app.ActivityThread.handleLaunchActivity(ActivityThre ad.java:2928)
at android.app.ActivityThread.-wrapl 1 (Unknown Source:0) at
android.app.ActivityThread$H.handleMessage(ActivityThread.
java:1 619)
at android.os.Handler.dispatchMessage(Handler.java:1 05)
at android.os.Looper.loop(Looper.java:1 64)
at android.app.ActivityThread.main(ActivityThread.java:
6695)
at java.lang.reflect. Method.invoke(Native Method) at
com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zy
gote.j ava:240)
at com.android.internal.os.Zygotelnit.main(Zygotelnit.java:
780)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method ‘java.lang.Object[] java. util.Collection.toArray’ on a null object reference
at java. util.ArrayList.addAll(ArrayList.java:588) at
cc.hurrypeng.fragmem.Util$FileHelper.getFragList(Util.java:
242)
at
cc. hurrypeng.fragmem. MainActivity.onCreate(MainActivity.jav a:78)
at android. app.Activity. performCreate(Activity.java:7054) at
a ndro id. a pp. Instru mentation. callActivityOnCreate(lnstrum ent ation.java:1 214)
at
android.app.ActivityThread. perform LaunchActivity(ActivityThr ead.java:2806)
9 more
```

**Comment by GrakePch** [comment]
Same here. You can install 0.4.1 and create at least one frags. Then update it to 0.5.0 to avoid the crash.:joy:

**Comment by KokeCacao** [comment]
Apparently, there is some problem here
```
public void getFragList(List<Frag> fragList) {
            String strJson = loadSDFile("frags.json");
            strJson = strJson.replaceAll(" ", "nbsp"); // Gson will drop dead when it meets a space in a json string
            List<Frag> newFragList = gson.fromJson(strJson, new TypeToken<List<Frag>>(){}.getType());
            fragList.clear();
            fragList.addAll(newFragList);
            for (Frag frag : fragList) {
                String title = frag.getTitle();
                title = title.replaceAll("nbsp", " ");
                frag.setTitle(title);
                String content = frag.getContent();
                content = content.replaceAll("nbsp", " ");
                frag.setContent(content);
            }
        }
```

`gson.fromJson(strJson, new TypeToken<List<Frag>>(){}.getType());` returns null, and `.addAll(null);` cause trouble...

A simple solution is:
```
List<Frag> newFragList = gson.fromJson(strJson, new TypeToken<List<Frag>>(){}.getType());
fragList.clear();
if (newFragList != null) fragList.addAll(newFragList);
```

@HurryPeng Correct me if I am wrong...

**Comment by HurryPeng** [comment]
@C6H5-NO2 A very detailed bug report!

**Comment by HurryPeng** [comment]
@KokeCacao Yes this is a better solution. In previous versions I used a variable in SharedPreferences to flag whether frags.json has been created, so that there should always be a frags.json. I thought they actually initialised no frag at all so I deleted those lines, forgetting that their creating a new file also counted. If I just fixed it in the old way, there would be another bug, that is, once frags.json is deleted manually, the program will never re-generate it, for it has been marked "initialised". Your way should be able to dodge that.
